db/compaction/compaction_job.cc

```diff
@@ -972,20 +979,29 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
         "CompactionJob::Run():PausingManualCompaction:2",
         reinterpret_cast<void*>(
             const_cast<std::atomic<bool>*>(manual_compaction_paused_)));
+    if (partitioner_ptr != nullptr) {
+      last_key_for_partitioner.assign(c_iter->user_key().data_,
```

Comment:
I still want to keep a `Reset()` method (or name it `OnOutputFileEnded()`) to signal new output file created. Also that way we don't need to pass last key to `ShouldPartition()`, thus saving a memcopy. What do you think?

Reply:
I was thinking about that and memcpy will not be noticeable compared to everything else. Keys are small. Most partitioners will have to keep last anyways so the copy will happen. Fixed partitioner can potentially copy less, but this makes everything much easier to read and maintain I think. We are doing this check for every pair of keys in compaction so that Reset was called same amount of times as ShouldPartition. Memcpy can be at least inlined.

When ShouldPartition return true, compaction will always split the file so there is no need to create event OnOutputFileEnded I think.

Reply:
I want to keep the `Reset()` for the following reasons:
1. currently SST split can cause by the partitioner and other conditions (e.g. `max_output_file_size`). I want to leave a mechanism notify the partitioner when SST split actually happen.
2. potentially it allow us to change API to make `ShouldPartition` just a hint.
3. reduce memcpy

Reply:
After a second thought I think the `Reset` method is not needed now and can be added later.